develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -50,6 +46,51 @@ example anatomy structure is outlined below:
     - `faces-config.xml`
     - `web.xml`
 
+## Auto-Generated WAB [](id=auto-generated-wab)
+
+The source of an auto-generated JSF Portlet WAB follows the standard
+directory layout for WAR projects. Bnd directives for the manifest can be placed in the liferay-plugin-package.properties file.  For use of Declarative Services in your JSF portlet, a java class that extends [GenericFacesPortlet](http://myfaces.apache.org/portlet-bridge/2.0/api/apidocs/javax/portlet/faces/GenericFacesPortlet.html) may be included and annotated with `@Component` on the class and `@Reference` on a method to inject a service.
+
+- `src/`
+    - `main/`
+        - `java/`
+        - `webapp/`
+            - `WEB-INF/`
+                - `faces-config.xml`
+                - ~~`liferay-display.xml`~~ (no longer needed)
+                - `liferay-plugin-package.properties`
```

Comment:
Right, but "no longer needed" needs to be qualified with "no longer needed if extending GenericFacesPortlet annotated with @Component" because you do need it for WARs that simply use GenericFacesPortlet.